api/v1alpha4/backstage_types.go

```diff
@@ -0,0 +1,330 @@
+
+	// List of references to Secrets objects mounted as extra files under the MountPath specified.
+	// For each item in this array, a key must be specified that will be mounted as a file.
+	// +optional
+	Secrets []FileObjectRef `json:"secrets,omitempty"`
+
+	// List of references to Persistent Volume Claim objects mounted as extra files
+	// For each item in this array, a key must be specified that will be mounted as a file.
+	// +optional
+	Pvcs []PvcRef `json:"pvcs,omitempty"`
+}
+
+type ExtraEnvs struct {
+	// List of references to ConfigMaps objects to inject as additional environment variables.
+	// For each item in this array, if a key is not specified, it means that all keys in the ConfigMap will be injected as additional environment variables.
+	// Otherwise, only the specified key will be injected as an additional environment variable.
+	// +optional
+	ConfigMaps []EnvObjectRef `json:"configMaps,omitempty"`
+
+	// List of references to Secrets objects to inject as additional environment variables.
+	// For each item in this array, if a key is not specified, it means that all keys in the Secret will be injected as additional environment variables.
+	// Otherwise, only the specified key will be injected as environment variable.
+	// +optional
+	Secrets []EnvObjectRef `json:"secrets,omitempty"`
+
+	// List of name and value pairs to add as environment variables.
+	// +optional
+	Envs []Env `json:"envs,omitempty"`
+}
+
+type EnvObjectRef struct {
+	// Name of the object
+	// We support only ConfigMaps and Secrets.
+	//+kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Key in the object
+	// +optional
+	Key string `json:"key,omitempty"`
+}
+
+type FileObjectRef struct {
+	// Name of the object
+	// Supported ConfigMaps and Secrets
+	//+kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Key in the object
+	// +optional
+	Key string `json:"key,omitempty"`
+
+	// Path to mount the Object. If not specified default-path/Name will be used
+	// +optional
+	MountPath string `json:"mountPath"`
+}
+
+type PvcRef struct {
+	// Name of the object
+	//+kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Path to mount PVC. If not specified default-path/Name will be used
+	// +optional
+	MountPath string `json:"mountPath"`
+}
+
+type Env struct {
+	// Name of the environment variable
+	//+kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Value of the environment variable
+	//+kubebuilder:validation:Required
+	Value string `json:"value"`
+}
+
+// BackstageStatus defines the observed state of Backstage
+type BackstageStatus struct {
+	// Conditions is the list of conditions describing the state of the runtime
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+// +operator-sdk:csv:customresourcedefinitions:displayName="Red Hat Developer Hub"
+
```

Comment:
I think it alpha4 has to be storage version as well, i.e. move
//+kubebuilder:storageversion
from alpha3